coding/trie_builder.hpp

```diff
@@ -39,25 +40,26 @@
 
 namespace trie
 {
-template <typename TSink, typename TChildIter, typename TValueList>
-void WriteNode(TSink & sink, TrieChar baseChar, TValueList const & valueList,
-               TChildIter const begChild, TChildIter const endChild, bool isRoot = false)
+template <typename TSink, typename TChildIter, typename TValueList, typename TSerializer>
+void WriteNode(TSink & sink, TSerializer const & serializer, TrieChar baseChar,
+               TValueList const & valueList, TChildIter const begChild, TChildIter const endChild,
+               bool isRoot = false)
 {
+  uint32_t const valueCount = valueList.Size();
   if (begChild == endChild && !isRoot)
   {
     // Leaf node.
-    valueList.Dump(sink);
+    valueList.Serialize(sink, serializer);
```

Comment:
There is some kind of leaky abstraction, as valueList.Serialize() does not serializes number of values.